cfme/generic_objects/definition/ui.py

```diff
@@ -0,0 +1,477 @@
+        return (
+            self.in_generic_object_definition and
+            self.title.text == 'Edit Generic Object Class'
+        )
+
+
+class GenericObjectDefinitionDetailsView(GenericObjectDefinitionView):
+    title = Text('#explorer_title_text')
+    configuration = Dropdown(text='Configuration')
+    summary = ParametrizedSummaryTable()
+    accordion = View.nested(AccordionForm)
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_generic_object_definition and
+            self.title.text == 'Generic Object Class {}'.format(self.context['object'].name)
+        )
+
+
+class GenericObjectAddButtonView(GenericObjectDefinitionView):
+    title = Text('#explorer_title_text')
+
+    button_type = BootstrapSelect(name='button_type')
+    name = Input(name='name')
+    description = Input(name='description')
+    display = BootstrapSwitch(name='display')
+    image = VersionPick({
+        Version.lowest(): BootstrapSelect('button_image'),
+        '5.9': FonticonPicker('button_icon')})
+    dialog = BootstrapSelect(name='dialog')
+    open_url = BootstrapSwitch(name='open_url')
+    display_for = BootstrapSelect(name='display_for')
+    submit_version = BootstrapSelect(name='submit_how')
+    system_process = BootstrapSelect(name='ae_instance')
+    system_message = Input(name='ae_message')
+    request = Input(name='request')
+    add_attribute_value_key = Button('Add Attribute/Value Pair')
+    attributes = ButtonParameterForm(param_type="Attribute")
+
+    add = Button('Add')
+    cancel = Button('Cancel')
+
+    @property
+    def is_displayed(self):
+        return self.title.text == 'Add a new Custom Button' and self.in_generic_object_definition
+
+
+class GenericObjectButtonGroupAddView(GenericObjectDefinitionView):
+    title = Text('#explorer_title_text')
+
+    name = Input(name='name')
+    display = BootstrapSwitch(name='display')
+    description = Input(name='description')
+    image = VersionPick({
+        Version.lowest(): BootstrapSelect('button_image'),
+        '5.9': FonticonPicker('button_icon')})
+
+    add = Button('Add')
+    cancel = Button('Cancel')
+
+    @property
+    def is_displayed(self):
+        return (
+            self.title.text == 'GenericObjectButtonGroupAddView' and
+            self.in_generic_object_definition
+        )
+
+    def after_fill(self, was_change):
+        # we need to click somewhere out side the form to get add button active,
+        # after icon is filled
+        if was_change:
+            self.browser.element('//body').click()
+
+
+class GenericObjectButtonGroupDetailsView(GenericObjectDefinitionView):
+    title = Text('#explorer_title_text')
+    configuration = Dropdown(text='Configuration')
+    basic_information = SummaryForm('Basic Information')
+    accordion = View.nested(AccordionForm)
+    table = Table('//h3[contains(text(), "Buttons")]/following-sibling::table')
+
+    @property
+    def is_displayed(self):
```

Comment:
Button groups have attributes like name and description, this does not look like a view class/method that will uniquely identify a button group instance, but _any_ button group details page.